optuna_dashboard/ts/components/PreferenceHistory.tsx

```diff
@@ -0,0 +1,212 @@
+    "::before": {},
+  }
+  if (type !== "none") {
+    cardComponentSx["::before"] = {
+      content: '""',
+      position: "absolute",
+      top: 0,
+      left: 0,
+      width: "100%",
+      height: "100%",
+      backgroundColor: theme.palette.mode === "dark" ? "white" : "black",
+      opacity: 0.2,
+      zIndex: 1,
+      transition: "opacity 0.3s ease-out",
+    }
+  }
+
+  return (
+    <Card
+      sx={{
+        width: trialWidth,
+        minHeight: trialHeight,
+        margin: theme.spacing(2),
+        padding: 0,
+      }}
+    >
+      <CardActions>
+        <Typography variant="h5">Trial {trial.number}</Typography>
+        <IconButton
+          sx={{
+            marginLeft: "auto",
+          }}
+          onClick={() => setDetailShown(true)}
+          aria-label="show detail"
+        >
+          <OpenInFullIcon />
+        </IconButton>
+      </CardActions>
+      <CardContent aria-label="trial" sx={cardComponentSx}>
+        <Box
+          sx={{
+            padding: theme.spacing(2),
+          }}
+        >
+          <MarkdownRenderer body={trial.note.body} />
+        </Box>
+
+        {type === "worst" ? (
+          <ClearIcon
+            sx={{
+              position: "absolute",
+              width: "100%",
+              height: "100%",
+              top: 0,
+              left: 0,
+              color: red[600],
+              zIndex: 1,
+              opacity: 0.3,
+              filter:
+                theme.palette.mode === "dark"
+                  ? "brightness(1.1)"
+                  : "brightness(1.7)",
+            }}
+          />
+        ) : null}
+      </CardContent>
+      <Modal open={detailShown} onClose={() => setDetailShown(false)}>
+        <Box
+          sx={{
+            position: "absolute",
+            top: 0,
+            left: 0,
+            right: 0,
+            bottom: 0,
+            width: "80%",
+            maxHeight: "90%",
+            margin: "auto",
+            overflow: "hidden",
+            backgroundColor: theme.palette.mode === "dark" ? "black" : "white",
+            borderRadius: theme.spacing(3),
+          }}
+        >
+          <Box
+            sx={{
+              width: "100%",
+              height: "100%",
+              overflow: "auto",
+            }}
+          >
+            <TrialListDetail
+              trial={trial}
+              isBestTrial={() => false}
+              directions={[]}
+              objectiveNames={[]}
+            />
+          </Box>
+        </Box>
+      </Modal>
+    </Card>
+  )
+}
+
+const ChoiceTrials: FC<{ choice: PreferenceChoice; trials: Trial[] }> = ({
+  choice,
+  trials,
+}) => {
+  const theme = useTheme()
+  const worst_trials = new Set([choice.clicked])
+
+  return (
+    <Box>
+      <Typography
+        variant="h6"
+        sx={{
+          marginBottom: theme.spacing(2),
+          fontWeight: theme.typography.fontWeightLight,
+        }}
+      >
+        {choice.timestamp.toISOString()}
```

Comment:
Could you use `formatDate()` instead?
https://github.com/optuna/optuna-dashboard/blob/main/optuna_dashboard/ts/dateUtil.ts